compiler/damlc/daml-opts/daml-opts-types/DA/Daml/Options/Types.hs

```diff
@@ -103,9 +102,8 @@ basePackages = ["daml-prim", "daml-stdlib"]
 mkOptions :: Options -> IO Options
 mkOptions opts@Options {..} = do
     mapM_ checkDirExists $ optImportPath <> optPackageDbs
-    mbDefaultPkgDb <- locateRunfilesMb (mainWorkspace </> "compiler" </> "damlc" </> "pkg-db")
-    let mbDefaultPkgDbDir = fmap (</> "pkg-db_dir") mbDefaultPkgDb
-    pkgDbs <- filterM Dir.doesDirectoryExist (toList mbDefaultPkgDbDir ++ [projectPackageDatabase])
+    defaultPkgDbDir <- locateRunfiles (mainWorkspace </> "compiler" </> "damlc" </> "pkg-db" </> "pkg-db_dir")
```

Comment:
Does this work on Windows if the default package db is not in the runfiles? I would have expected that this fails to return a location.

Reply:
I took a look at the generated manifest file and it had an entry for `pkg-db_dir`.

Reply:
Even for `damlc_bootstrap`? There definitely should be one for `damlc` but we had the case for the non-exist pkg db dir for bootstrapping purposes afaik.

Reply:
No, but for `damlc-bootstrap` there's no entry for `pkg-db` either way. I.e. neither `compiler/damlc/pkg-db` nor `compiler/damlc/pkg-db/pkg-db_dir` in neither `.runfiles_manifest` nor `.runfiles`.

Reply:
What I don’t understand is why this is not an error. What does `locateRunfiles` return if the directory does not exist?

Reply:
`rlocation` does not check if the path actually exists. This is also following the Bazel spec:

> Directory-based: appends the runfile's path to the runfiles root. The client is responsible for checking that the resulting path exists.

Tracing the result of `locateRunfiles` when building `//compiler/damlc/pkg-db:daml-prim-1.5` shows that the `pkg-db` path indeed does not exist. But, as long as it's not read that's not an error.

Reply:
The reason why I am confused is that I don’t understand how we can produce any reasonable path existing or not on Windows since the paths there are read from the manifest instead of simply concatenated. Sounds like I am misunderstanding how things work here?

Reply:
I see, yes, on Windows the fallback is to just return the original path, which will typically not exist. It's not pretty, but it seems like the most consistent solution across Windows and Unix. It doesn't cause an error down the line in this case, because of the check if the directory exists in the following line.

Reply:
Thank you for the explanation! Now I understand what’s going on.

Reply:
Turns out this is problematic with `locateRunfiles` in the `package_app` case. I misunderstood `locateRunfiles`, in the `package_app` case it is meant to just return the `resources` directory and the use-site then knows what relative path to append to find the actual runfiles. In the non-`package_app` case `locateRunfiles` falls back to bazel-runfiles to find the corresponding directory.

I've changed things to match this behavior. On the bright side it reduces the code even further.